import-mailbox-to-gmail.py

```diff
@@ -159,6 +189,39 @@ def get_label_id_from_name(service, username, labels, labelname):
     raise
 
 
+# TODO: instead of passing mbox_label_name as param, can't we look it up from id?
+def get_metadata(service, username, labels, msg, mbox_label_id, mbox_label_name):
+  """Find Gmail labels and preserve them for import.
+
+  Returns:
+    A metadata object.
+  """
+  label_ids = [mbox_label_id]
+
+  if args.takeout_labels and 'X-Gmail-Labels' in msg:
+    gmail_labels = next(reader([msg['X-Gmail-Labels'].replace('\r\n', '')]))
+    logging.info('Found Gmail Labels: %s', gmail_labels)
```

Comment:
Maybe "Found Gmail labels from Takeout"? Otherwise the user might not understand the context (if they don't remember this mbox came from Takeout, etc.), this would help differentiate it from finding a label in the sense of finding a new mbox file.